app/code/Magento/InventoryInStorePickupFrontend/view/frontend/web/js/model/pickup-locations-service.js

```diff
@@ -0,0 +1,227 @@
+                {},
+                addressConverter.formAddressDataToQuoteAddress({
+                    firstname: location.name,
+                    lastname: 'Store',
+                    street: location.street,
+                    city: location.city,
+                    postcode: location.postcode,
+                    country_id: location.country_id,
+                    telephone: location.telephone,
+                    region_id: location.region_id,
+                    save_in_address_book: 0,
+                    custom_attributes: {
+                        sourceCode: location.source_code,
+                    },
+                }),
+                {
+                    canUseForBilling: function() {
+                        return false;
+                    },
+                    getType: function() {
+                        return 'store-pickup-address';
+                    },
+                }
+            );
+
+            this.selectedLocation(location);
+            selectShippingAddressAction(address);
+            checkoutData.setSelectedShippingAddress(address.getKey());
+        },
+        /**
+         * Formats address returned by REST endpoint to match checkout address field naming.
+         * @param {object} address Address object returned by REST endpoint.
+         */
+        formatAddress: function(address) {
+            return {
+                name: address.name,
+                description: address.description,
+                latitude: address.latitude,
+                longitude: address.longitude,
+                street: [address.street],
+                city: address.city,
+                postcode: address.postcode,
+                country_id: address.country_id,
+                country: this.getCountryName(address.country_id),
+                telephone: address.phone,
+                region_id: address.region_id,
+                region: this.getRegionName(
+                    address.country_id,
+                    address.region_id
+                ),
+                source_code: address.source_code,
+            };
+        },
+        /**
+         * @param {*} countryId
+         * @return {String}
+         */
+        getCountryName: function(countryId) {
+            return countryData()[countryId] != undefined
+                ? countryData()[countryId].name
+                : ''; //eslint-disable-line
+        },
+        /**
+         * Returns region name based on given country and region identifiers.
+         * @param {string} countryId Country identifier.
+         * @param {string} regionId Region identifier.
+         */
+        getRegionName: function(countryId, regionId) {
+            var regions = countryData()[countryId]
+                ? countryData()[countryId].regions
+                : null;
+
+            return regions && regions[regionId] ? regions[regionId].name : '';
+        },
+        processError: function(response) {
```

Comment:
Use inventory/app/code/Magento/Checkout/view/frontend/web/js/model/error-processor.js instead of this method.